aws-lc-rs/src/ec.rs

```diff
@@ -228,14 +228,16 @@ pub(crate) fn marshal_public_key(ec_key: &ConstPointer<EC_KEY>) -> Result<Public
 pub(crate) unsafe fn ec_key_from_public_point(
     ec_group: &LcPtr<EC_GROUP>,
     public_ec_point: &LcPtr<EC_POINT>,
-) -> Result<DetachableLcPtr<EC_KEY>, Unspecified> {
-    let ec_key = DetachableLcPtr::new(EC_KEY_new())?;
+) -> Result<LcPtr<EC_KEY>, Unspecified> {
+    let nid = EC_GROUP_get_curve_name(ec_group.as_const_ptr());
+    let ec_key = LcPtr::new(EC_KEY_new())?;
     if 1 != EC_KEY_set_group(*ec_key, **ec_group) {
         return Err(Unspecified);
     }
     if 1 != EC_KEY_set_public_key(*ec_key, **public_ec_point) {
         return Err(Unspecified);
     }
+    validate_ec_key(&ec_key.as_const(), nid)?;
```

Comment:
should you then add validation to `ec_key_from_public_private`?

Reply:
I agree, the validation is not invoked inside `ec_key_from_public_private`. (The function was only used [one place](https://github.com/aws/aws-lc-rs/blob/e8632a4f80ad059e644b23e6e80c7988a4b7646a/aws-lc-rs/src/ec/key_pair.rs#L159) and was making its own validation call.)